distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Writer.enso

```diff
@@ -31,18 +30,18 @@ make_java_existing_data_mode on_existing_file match_columns = case on_existing_f
    Writes a Table to an Excel file.
 
    Arguments:
-write_file : File -> Table -> Existing_File_Behavior -> (Sheet | Cell_Range) -> (Boolean|Infer) -> Match_Columns -> Problem_Behavior -> Boolean
+write_file : File -> Table -> Existing_File_Behavior -> Excel_Section -> (Boolean|Infer) -> Match_Columns -> Problem_Behavior -> Boolean
 write_file file table on_existing_file section headers match_columns _ xls_format=False =
     workbook = if file.exists.not then ExcelWriter.createWorkbook xls_format else
-        handle_reader file stream->(ExcelReader.getWorkbook stream xls_format)
+        Excel_Reader.handle_reader file stream->(ExcelReader.getWorkbook stream xls_format)
 
     existing_data_mode = make_java_existing_data_mode on_existing_file match_columns
-    java_headers = make_java_headers headers
+    java_headers = Excel_Reader.make_java_headers headers
     ExcelWriter.setEnsoToTextCallbackIfUnset (.to_text)
     result = handle_writer <| case section of
-        Worksheet sheet skip_rows row_limit ->
+        Excel_Section.Worksheet sheet skip_rows row_limit ->
             ExcelWriter.writeTableToSheet workbook sheet existing_data_mode skip_rows table.java_table row_limit java_headers
-        Cell_Range address skip_rows row_limit -> case address of
+        Excel_Section.Cell_Range address skip_rows row_limit -> case address of
             Excel_Range.Value java_range -> ExcelWriter.writeTableToRange workbook java_range existing_data_mode skip_rows table.java_table row_limit java_headers
             _ : Text -> ExcelWriter.writeTableToRange workbook address existing_data_mode skip_rows table.java_table row_limit java_headers
```

Comment:
Once the type is just `Excel_Section` (ofc with the new language spec it has to be that - so the change was needed), we should handle any parameter of the given type.

I.e. IMO we should never throw `Inexhaustive_Pattern_Match_Error` (currently this will happen for `Excel_Section.Range_Names`, for example) for a type that is _valid_. Instead, I think what would be a good way would be to add a catchall:
```
_ : Excel_Section -> Error.throw (Illegal_Argument.Error "Only a Worksheet or Cell_Range is allowed in write_file"`
```

This is a bit of a matter of style, but IMO stuff like `Inexhaustive_Pattern_Match_Error` or `Type_Error` should indicate that a wrong type was given (i.e. situations that should essentially be prevented by a typechecker). Whereas for situations where the type is OK but something is not supported the less-internal more 'user friendly' `Illegal_Argument.Error` should be thrown instead.

Does that sound reasonable?